### PR TITLE
[P2T support No.18-19] [BIT] __mul__, __sub__, full, fused_feedforward, fused_layer_norm

### PR DESCRIPTION
**Pull request by Cutelemon6**
## paddle.Tensor.\_\_mul__

测试了一条 #88 中遗留的大 size tensor，测试通过，移入配置到 support2torch_6.txt

```python
paddle.Tensor.__mul__(Tensor([2, 1, 32768, 32768],"float16"), 10000.0, )
```

## paddle.Tensor.\_\_sub__

测试了一条 #88 中遗留的大 size tensor，测试通过，移入配置到 support2torch_6.txt

```python
paddle.Tensor.__sub__(Tensor([2, 1, 32768, 32768],"float16"), 1, )
```

## paddle.full

测试了一条 #122 中提到的可能导致大 size tensor 的配置 ，设置第一个位置 tensor （即 shape）初始化每个元素在 [0, 32)

```python
paddle.full(shape=Tensor([5], "int64"), fill_value=-2)
```

配置通过，移入 support2torch_6.txt

## paddle.incubate.nn.functional.fused_feedforward

没有直接对应 fuse api，模拟 api 文档说明使用 torch api 组合实现。转换细节：

1. paddle 的线性层和 torch 的线性层输入参数格式不一样（torch: [out_features, in_features], paddle: [in_features, out_features]），torch 进行计算前需要先对 weight 最后两维进行转置。

   ```python
   linear1_weight = linear1_weight.transpose(-2, -1)
   linear2_weight = linear2_weight.transpose(-2, -1)
   ```

2. mode == 'upscale_in_train' 的 dropout 有 torch 实现，mode == 'down_scale_in_infer' 需要手写实现

   ```python
   dropout = torch.nn.functional.dropout if mode == 'upscale_in_train' else lambda x, p, training: (
       x * torch.bernoulli(torch.full(x.shape, 1 - p)).to(x.device) if training else x * (1 - p)
   )
   ```

组合实现逻辑：

```python
residual = src;
if pre_layer_norm:
    src = layer_norm(src)
src = linear(dropout(activation(linear(src))))
src = residual + dropout(src)
if not pre_layer_norm:
    src = layer_norm(src)
```

20 个中 4 个输入 dtype 不同，采用 --test_amp=True 测试，3 个报精度错误，日志放入 fresh_report，配置放入 accuracy_error_2.txt，1 个通过；剩下 16 个中 3 个报精度错误，日志放入 fresh_report，配置放入 accuracy_error_2.txt，13 个通过。通过配置放入 support2torch_6.txt。

## paddle.incubate.nn.functional.fused_layer_norm

没有直接对应 api，使用 torch api 组合实现

```python
x_rb = x + residual * residual_alpha + bias
out_residual = x_rb
# flatten norm dims
x_rb = torch.flatten(x_rb, start_dim=begin_norm_axis)
out_mean = torch.mean(x_rb, dim=-1).reshape(-1)
out_var = torch.var(x_rb, dim=-1, unbiased=False).reshape(-1)

if norm_weight is not None or norm_bias is not None:
    if norm_weight is None:
        norm_weight = torch.ones(x_rb.shape[-1])
    if norm_bias is None:
        norm_bias = torch.zeros(x_rb.shape[-1])
    x_rb = torch.nn.functional.layer_norm(x_rb, [x_rb.shape[-1]], weight=norm_weight, bias=norm_bias, eps=epsilon)

x = torch.reshape(x_rb, x.shape)
```

量化部分参考 paddle 库源代码 /paddle/test/legacy_test/test_fused_layernorm_op.py 中的 quant_helper()

```python
x = quant_scale * x * quant_max_bound
# using banker's rounding
if quant_round_type == 0:
    x = torch.round(x)
else: #  Round to nearest if type != 0
    x = torch.floor(x + 0.5)
x = torch.clamp(x, min=quant_min_bound, max=quant_max_bound).to(torch.int8)
```

75 个用例中 27 个需采用 --test_amp=True 测试，其中 13 个通过，14个报 accuracy error，日志放入 
 eport/fresh_report/accuracy/，配置放入 accuracy_error_2.txt。剩余 48 个用例中 1 个通过，47 个报 accuracy error，日志放入 report/fresh_report/accuracy/，配置放入 accuracy_error_2.txt。通过配置放入 support2torch_6.txt。上述 accuracy error 均表现为 shape mismatch，如 `(91, 64, 0), (91, 64) mismatch`，这是由于 paddle api 输出的结果包含了 0-size tensor (uninitialized tensor) 导致的（paddle api 输入关键字参数 residual == None 会导致这个问题），而模拟的 torch 组合逻辑能正确输出。

另外添加了三条会导致 paddle api 抛出 cuda 700 error 的配置，分别是输入关键字参数 bias, norm_weight, norm_bias 为 None 时会导致 paddle api 抛出 cuda 700 error，将配置放入 merged_not_support, 日志放入 fresh_report/arruracy

- https://github.com/PaddlePaddle/Paddle/issues/72643

### CODE REVIEW

File: tester/api_config/api_config_merged_not_support.txt [hunk 1]
```diff
@@ -28,4 +28,7 @@ paddle.cast(Tensor([120, 84],"float32"), "float8_e4m3fn", )
...
 paddle.cast(Tensor([6, 1, 3, 3],"float32"), "float8_e4m3fn", )
-paddle.cast(Tensor([576, 120],"float32"), "float8_e4m3fn", )
\ No newline at end of file
+paddle.cast(Tensor([576, 120],"float32"), "float8_e4m3fn", )
+paddle.incubate.nn.functional.fused_layer_norm(Tensor([16, 256],"float32"), Tensor([256],"float32"), Tensor([256],"float32"), 1e-05, begin_norm_axis=1, bias=None, residual=Tensor([16, 256],"float32"), residual_alpha=0.69204696, quant_scale=127, quant_round_type=0, quant_max_bound=127, quant_min_bound=-128)
+paddle.incubate.nn.functional.fused_layer_norm(Tensor([16, 256],"float32"), None, Tensor([256],"float32"), 1e-05, begin_norm_axis=1, bias=Tensor([256],"float32"), residual=Tensor([16, 256],"float32"), residual_alpha=0.69204696, quant_scale=127, quant_round_type=0, quant_max_bound=127, quant_min_bound=-128)
+paddle.incubate.nn.functional.fused_layer_norm(Tensor([16, 256],"float32"), Tensor([256],"float32"), None, 1e-05, begin_norm_axis=1, bias=Tensor([256],"float32"), residual=Tensor([16, 256],"float32"), residual_alpha=0.69204696, quant_scale=127, quant_round_type=0, quant_max_bound=127, quant_min_bound=-128)
```
Comment (cszdrg) [inline tester/api_config/api_config_merged_not_support.txt:34]: 这些api是任务一都过不了吗？

Comment (Cutelemon6) [reply tester/api_config/api_config_merged_not_support.txt:34]: 试了一下确实是同样的 700 error 过不了

Comment (cszdrg) [reply tester/api_config/api_config_merged_not_support.txt]: 那把配置放在任务一失败的文件里吧

**Review by cszdrg — APPROVED** [review]

**Review by wanghuancoder — APPROVED** [review]
LGTM
